polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -91,6 +101,44 @@ async fn send_response(stream: &mut UnixStream, result: PrepareResult) -> io::Re
 	framed_send(stream, &result.encode()).await
 }
 
+fn start_memory_tracking(fd: RawFd, limit: Option<isize>) {
+	unsafe {
+		// SAFETY: Inside the failure handler, the allocator is locked and no allocations or
+		// deallocations are possible. For Linux, that always holds for the code below, so it's
+		// safe. For MacOS, that technically holds at the time of writing, but there are no future
+		// guarantees.
+		// The arguments of unsafe `libc` calls are valid, the payload validity is covered with
+		// a test.
+		ALLOC.start_tracking(
+			limit,
+			Some(Box::new(move || {
+				#[cfg(target_os = "linux")]
+				{
+					// Syscalls never allocate or deallocate, so this is safe.
+					libc::syscall(libc::SYS_write, fd, OOM_PAYLOAD.as_ptr(), OOM_PAYLOAD.len());
```

Comment:
Why do we need raw sys calls here?

Reply:
Because we cannot have allocations ... but 1. I don't see why those functions should should ever allocate, 2. why would this be a problem? Can we make it a non-problem?

Reply:
@eskimor There's no guarantee that they don't allocate/deallocate. Although you're right that it's probably unlikely that it will be a problem.

(For example, `exit` will call `atexit` handlers, which most certainly *can* allocate/deallocate, and anyone from anywhere can register an `atexit` handler. Or one of the dependencies could interpose one of the symbols and allocate from there; e.g. the `crash-handler` crate which was used in a previous version of this patch actually does something like that, so it's not completely impossible.)